### ISSUE TITLE
[Question] How to add modules in Ubuntu 16.04

### ISSUE THREAD

**Issue by Endogen** [cockpit-project/cockpit#7856]
Is there a way to add additional modules in Ubuntu 16.04? Right from the start i have (translated from german):
- System
- Logs
- Storage
- Network
- Container
- Accounts
- Services
- Terminal

**Comment by martinpitt** [comment]
Please be more specific what you are looking for. For example, you can install `cockpit-packagekit` to get "Software Updates", or `cockpit-machines` to manage libvirt VMs. You can also write and use your own pages, using the [API](http://cockpit-project.org/guide/latest/); see e. g. the [pinger blog post](http://cockpit-project.org/blog/creating-plugins-for-the-cockpit-user-interface.html) or [the examples](https://github.com/cockpit-project/cockpit/tree/master/examples).

**Comment by Endogen** [comment]
Ok, then the question is, which modules are available and how do i install them (meaning, what is the package name). Is there a list somewhere?

**Comment by Endogen** [comment]
Is that the complete list?
```
cockpit-bridge - Cockpit bridge server-side component
cockpit-dashboard - Cockpit SSH remoting and dashboard
cockpit-doc - Cockpit deployment and developer guide
cockpit-docker - Cockpit user interface for Docker containers
cockpit-machines - Cockpit user interface for virtual machines
cockpit-networkmanager - Cockpit user interface for networking
cockpit-storaged - Cockpit user interface for storage
cockpit-system - Cockpit admin interface for a system
cockpit-tests - Tests for Cockpit
cockpit-ws - Cockpit Web Service
cockpit-pcp - Cockpit PCP integration
cockpit-shell - Cockpit Shell user interface package
cockpit-packagekit - Cockpit user interface for packages
```

**Comment by martinpitt** [comment]
Yes, that looks complete. You can get it in various ways, like `apt search cockpit` or `apt showsrc cockpit` and look at `Package-List`, or on Launchpad: https://launchpad.net/ubuntu/+source/cockpit

**Comment by q2dg** [comment]
cockpit-pcp is no longer packaged in artful or bionic Ubuntu versions.
In fact, in https://launchpad.net/ubuntu/+source/cockpit it doesn't appear, either.
Do you know why??
Thanks!

**Comment by martinpitt** [comment]
pcp is in a rather sorry [state](https://tracker.debian.org/pkg/pcp) in Debian, has been for several years, and also not exactly healthy upstream, so we can't have cockpit-pcp in Debian. It has been too cumbersome to build cockpit-pcp on Ubuntu only, as then the packages cannot be in sync any more and need to be maintained separately. Also, pcp isn't maintained in Ubuntu either, just taken as-is from Debian.

The bottom line is that the feature is not important enough and too brittle, so it's not currently supported on Debian/Ubuntu.

**Comment by q2dg** [comment]
Oooh, thanks a lot for your explanation.
What a pity :-(

**Comment by DavidFW1960** [comment]
That's a shame particularly seeing as the Cockpit GUI has a link in System to "enable persistent metrics" and that wants to install cockpit-pcp which it can't find a source for. Otherwise loving cockpit though.

**Comment by martinpitt** [comment]
@DavidFW1960 :  cockpit-pcp has been available in Ubuntu (16.04, 18.04, and 18.10) for quite some time now. E. g. on 16.04:

```
 curl -s http://archive.ubuntu.com/ubuntu/dists/xenial-backports/universe/binary-amd64/Packages.xz | xz -d| grep -A10 '^Package: cockpit-pcp'
Package: cockpit-pcp
Architecture: amd64
Version: 178-1~ubuntu16.04.1
Priority: optional
Section: universe/admin
Source: cockpit
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Utopia Maintenance Team <pkg-utopia-maintainers@lists.alioth.debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 233
```

**Comment by DavidFW1960** [comment]
I'm on Debian Stretch. I had to install backports to even use cockpit but no sources for cockpit-pcp

**Comment by DavidFW1960** [comment]
ok... so I should be able to install it when you fix it to build then?

**Comment by martinpitt** [comment]
Not cockpit-pcp (as I said, pcp isn't in stretch, so cockpit-pcp cannot be built there either), but surely the other cockpit modules. Although they shoud all be in version 172 already, just at older versions.

**Comment by DavidFW1960** [comment]
> , but at least I should fix the package to build.

So I guess I don't understand what you meant by this then. 

The only thing I have found that doesn't work is enabling persistent metrics.. Are you saying you are going to rebuild cockpit to remove this option on Debian?

**Comment by martinpitt** [comment]
For now I just fixed the recent upload to build at all again. Not offering the install option in that case should be done as well indeed, but I haven't done that yet.
